application_manager/plugins/spark_mesos/plugin.py

```diff
@@ -0,0 +1,250 @@
+configure_logging()
+
+
+class SparkMesosProvider(base.PluginInterface):
+
+    def __init__(self):
+        self.get_frameworks_url = "%s:%s" % (api.mesos_url,
+                                             api.mesos_port)
+        self.app_id = "app-spark-mesos-" + str(uuid4())[:8]
+
+
+    def get_title(self):
+        return 'Spark-Mesos on Open Nebula plugin for BigSea framework'
+
+    def get_description(self):
+        return 'It runs an spark application on a Mesos cluster'
+
+    def to_dict(self):
+        return {
+            'name': self.name,
+            'title': self.get_title(),
+            'description': self.get_description(),
+        }
+
+    def execute(self, data):
+        binary_url = str(data['binary_url'])
+        execution_class = str(data['execution_class'])
+        execution_parameters = str(data['execution_parameters'])
+        expected_time = int(data['expected_time'])
+        number_of_jobs = int(data['number_of_jobs'])
+        starting_cap = 0 if data['starting_cap'] == "" \
+                         else int(data['starting_cap'])
+
+        actuator = str(data['actuator'])
+
+        plugin_log.log("%s | Submission id: %s" % (time.strftime("%H:%M:%S"),
+                                                   self.app_id))
+
+        plugin_log.log("%s | Connecting with Mesos cluster..." %
+                      (time.strftime("%H:%M:%S")))
+
+        conn = self._get_ssh_connection(api.mesos_url,
+                                        api.cluster_username,
+                                        api.cluster_password,
+                                        api.cluster_key_path)
+
+        plugin_log.log("%s | Connected with Mesos cluster" %
+                      (time.strftime("%H:%M:%S")))
+
+        # Execute all the spark needed commands
+        # to run an spark job from command line
+        if execution_class != "" and execution_class is not None:
+
+            # If the class field is empty, it means that the
+            # job binary is python
+            binary_path = '~/exec_bin.jar'
+            spark_run = ('sudo %s --name %s ' +
+                              '--master mesos://%s:%s ' +
+                              '--class %s %s %s')
+        else:
+            binary_path = '~/exec_bin.py'
+            spark_run = ('sudo %s --name %s ' +
+                              '--master mesos://%s:%s ' +
+                              '%s %s %s')
+
+        plugin_log.log("%s | Download the binary to cluster" %
+                      (time.strftime("%H:%M:%S")))
+
+        try:
+            stdin, stdout, stderr = conn.exec_command('wget %s -O %s' %
+                                                      (binary_url,
+                                                       binary_path))
+
+            plugin_log.log("%s | Waiting for download the binary..."
+                            % (time.strftime("%H:%M:%S")))
+
+            stdout.read()
```

Comment:
Does stdout.read() raise an exception when wget fails?

Reply:
It's an unusual error. Anyway, this is **TODO: Fix possible wget error**